rhino/src/main/java/org/mozilla/javascript/Hashtable.java

```diff
@@ -123,20 +123,25 @@ public int size() {
     public void put(Object key, Object value) {
         final Entry nv = new Entry(key, value);
 
-        if (!map.containsKey(nv)) {
-            // New value -- insert to end of doubly-linked list
-            map.put(nv, nv);
-            if (first == null) {
-                first = last = nv;
-            } else {
-                last.next = nv;
-                nv.prev = last;
-                last = nv;
-            }
-        } else {
-            // Update the existing value and keep it in the same place in the list
-            map.get(nv).value = value;
-        }
+        map.compute(
```

Comment:
we will save one hash computation, but have the price of an additional invocation of a lambda expression:
I did this short test:
```java
	@Test
	public void speed() {
		for (long i = 0L; i < 10_000_000L; i++) {
			ht.put("one", 1);
			ht.put("two", 2);
			ht.put("three", 3);
			ht.put("four", 4);
			ht.put("five", 5);
			ht.put("six", 6);
			ht.put("seven", 7);
			ht.put("eight", 8);
			ht.put("nine", 9);
			ht.put("ten", 10);

		}
	}
```
it runs with the old code in apporx 8,5sec and with the new code approx 5,5 sec